plugins/repository-s3/src/main/java/org/elasticsearch/cloud/aws/InternalAwsS3Service.java

```diff
@@ -156,7 +158,17 @@ public static AWSCredentialsProvider buildCredentials(Logger logger, Deprecation
             }
         }
 
-        return credentials;
+        return new AWSCredentialsProvider() {
+            @Override
+            public AWSCredentials getCredentials() {
+                return SocketAccess.doPrivileged(credentials::getCredentials);
```

Comment:
We only need this for the instance profile credentials. The rest are read and created statically, so definitely do not require socket access.